File: pilhas-filas/javascript/stack2.js
export default class Stack2 {
    constructor() {
        this.list = [];
    }

    isEmpty() {
        return this.list.length === 0;
    }

    push(item) {
        this.list.push(item);
    }

    pop() {
        if (this.isEmpty()) {
            throw new Error("Stack is empty");
        }
        return this.list.pop();
    }

    peek() {
        if (this.isEmpty()) {
            throw new Error("Stack is empty");
        }
        return this.list[this.list.length - 1];
    }

    count() {
        return this.list.length;
    }

    clear() {
        this.list = [];
    }
}

File: pilhas-filas/java/src/Stack2.java
import java.util.ArrayList;
import java.util.List;

public class Stack2<T> {
    private List<T> list;

    public Stack2() {
        this.list = new ArrayList<>();
    }

    public boolean isEmpty() {
        return this.list.isEmpty();
    }

    public void push(T item) {
        this.list.add(item);
    }

    public T pop() {
        if (this.isEmpty()) {
            throw new IllegalStateException("Stack is empty");
        }
        return this.list.remove(this.list.size() - 1);
    }

    public T peek() {
        if (this.isEmpty()) {
            throw new IllegalStateException("Stack is empty");
        }
        return this.list.get(this.list.size() - 1);
    }

    public int count() {
        return this.list.size();
    }
    
    public void clear() {
        this.list = new ArrayList<>();
    }
}
